src/Data/Ord/Linear.hs

```diff
@@ -0,0 +1,62 @@
+{-# LANGUAGE FlexibleInstances #-}
+{-# LANGUAGE LinearTypes #-}
+{-# LANGUAGE MonoLocalBinds #-}
+{-# LANGUAGE NoImplicitPrelude #-}
+{-# LANGUAGE UndecidableInstances #-}
+
+
+module Data.Ord.Linear
+  ( Ord(..)
+  )
+  where
+
+import Data.Eq.Linear
+import qualified Prelude as Unrestricted
+import Prelude (Bool(..))
+import Data.Ord (Ordering(..))
+import qualified Unsafe.Linear as Unsafe
+
+class Eq a => Ord a where
+  {-# MINIMAL compare | (<=) #-}
+  compare :: a #-> a #-> Ordering
+  compare = Unsafe.toLinear2 compare'
+    where
+      compare' :: Ord a => a -> a -> Ordering
+      compare' x y
+        | x == y = EQ
+        | x <= y = LT
+        | True = GT
+
+  (<=) :: a #-> a #-> Bool
+  x <= y = fromCompare (compare x y)
+    where
+      fromCompare :: Ordering #-> Bool
+      fromCompare LT = True
+      fromCompare EQ = True
+      fromCompare GT = False
+
+  (<) :: a #-> a #-> Bool
+  x < y = compare x y == LT
+  (>) :: a #-> a #-> Bool
+  x > y = compare x y == GT
+  (>=) :: a #-> a #-> Bool
+  x >= y = compare x y /= LT
+
+  max :: a #-> a #-> a
+  max = Unsafe.toLinear2 max'
+    where
+      max' :: Ord a => a -> a -> a
+      max' x y = case compare x y of
+        GT -> x
+        _ -> y
+
+  min :: a #-> a #-> a
+  min = Unsafe.toLinear2 min'
+    where
+      min' :: Ord a => a -> a -> a
+      min' x y = case compare x y of
+        LT -> x
+        _ -> y
+
+instance Unrestricted.Ord a => Ord a where
+  compare = Unsafe.toLinear2 Unrestricted.compare
```

Comment:
This is very wrong. We could give some (unsafe!) deriving-via combinator to that effect. But let's do that later: add an issue instead.

```suggestion
```